.github/workflows/android.yml

```diff
@@ -77,6 +78,49 @@ jobs:
           if-no-files-found: ignore
           path: ${{ runner.temp }}/artifacts/
 
+  # Running Android emulator directly on the runner and not using Docker
+  run-emulator:
+    needs: build-llm-demo
+    runs-on: amz2023.linux.4xlarge
+    env:
+      ANDROID_NDK_VERSION: r26c
+    steps:
+      - name: Setup SSH (Click me for login details)
+        uses: pytorch/test-infra/.github/actions/setup-ssh@main
+        if: ${{ !contains(matrix.runner, 'gcp.a100') }}
+        with:
+          github-secret: ${{ secrets.GITHUB_TOKEN }}
+          instructions: |
+            All testing is done inside the container, to start an interactive session run:
+              docker exec -it $(docker container ps --format '{{.ID}}') bash
+
+      - uses: actions/checkout@v3
+        with:
+          submodules: false
+
+      - name: Setup conda
+        uses: pytorch/test-infra/.github/actions/setup-miniconda@main
+        with:
+          python-version: '3.10'
+
+      - name: Install Android dependencies
+        shell: bash
+        run: |
+          set -eux
+
+          # Reuse the script that install Android on ET Docker image
+          sudo -E bash .ci/docker/common/install_android.sh
+
+      # NB: It takes about 10m to cold boot the emulator here
```

Comment:
This is huge

Reply:
Is the emulator configurable? Like the RAM allocated to it. I know typically we would only use it to validate storiesM, but if it's configurable, it can be potentially used on a non-critical path to validate large GenAI model first before actual benchmarking on device.

Reply:
I think it has something to do with our Linux runner, it should be faster when hardware accelerator is support according to https://github.com/ReactiveCircus/android-emulator-runner.  Let me try to use GitHub runner and compare the perf.

Reply:
https://github.com/ReactiveCircus/android-emulator-runner?tab=readme-ov-file#configurations


ram-size maybe 12GB?

Also           emulator-options: -no-snapshot-save -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
          disable-animations: true
might be helpful

Reply:
Yeah, it's worth to try out these options, the doc also mentions some caching setup that could make it faster. Quouting https://github.com/ReactiveCircus/android-emulator-runner

```
We can significantly reduce emulator startup time by setting up AVD snapshot caching:

add a gradle/actions/setup-gradle@v3 step for caching Gradle, more details see https://github.com/ReactiveCircus/android-emulator-runner/issues/229
add an actions/cache@v4 step for caching the avd
add a reactivecircus/android-emulator-runner@v2 step to generate a clean snapshot - specify emulator-options without no-snapshot
add another reactivecircus/android-emulator-runner@v2 step to run your tests using existing AVD / snapshot - specify emulator-options with no-snapshot-save
```

Reply:
Thank you! Current cold startup time is slower than on device testing